ERROR:src:405 Method Not Allowed: The method is not allowed for the requested URL.
Traceback (most recent call last):
  File "D:\Development\full-stack-2024\flask-rest-api\.venv\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Development\full-stack-2024\flask-rest-api\.venv\Lib\site-packages\flask\app.py", line 854, in dispatch_request
    self.raise_routing_exception(req)
  File "D:\Development\full-stack-2024\flask-rest-api\.venv\Lib\site-packages\flask\app.py", line 463, in raise_routing_exception
    raise request.routing_exception  # type: ignore[misc]
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Development\full-stack-2024\flask-rest-api\.venv\Lib\site-packages\flask\ctx.py", line 362, in match_request
    result = self.url_adapter.match(return_rule=True)  # type: ignore
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Development\full-stack-2024\flask-rest-api\.venv\Lib\site-packages\werkzeug\routing\map.py", line 624, in match
    raise MethodNotAllowed(valid_methods=list(e.have_match_for)) from None
werkzeug.exceptions.MethodNotAllowed: 405 Method Not Allowed: The method is not allowed for the requested URL.
ERROR:src:415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.
Traceback (most recent call last):
  File "D:\Development\full-stack-2024\flask-rest-api\.venv\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Development\full-stack-2024\flask-rest-api\.venv\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Development\full-stack-2024\flask-rest-api\src\controllers\user_controller.py", line 17, in authenticate
    data = request.get_json()
           ^^^^^^^^^^^^^^^^^^
  File "D:\Development\full-stack-2024\flask-rest-api\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 604, in get_json
    return self.on_json_loading_failed(None)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Development\full-stack-2024\flask-rest-api\.venv\Lib\site-packages\flask\wrappers.py", line 131, in on_json_loading_failed
    return super().on_json_loading_failed(e)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Development\full-stack-2024\flask-rest-api\.venv\Lib\site-packages\werkzeug\wrappers\request.py", line 647, in on_json_loading_failed
    raise UnsupportedMediaType(
werkzeug.exceptions.UnsupportedMediaType: 415 Unsupported Media Type: Did not attempt to load JSON data because the request Content-Type was not 'application/json'.
